### PR TITLE
[32584] start merging xdruple-server contents

### PR DESCRIPTION
**Pull request by gilmoskowitz**
also:
- fix misc bugs as they manifested
- consolidate duplicated code
- address coding standard issues

### CODE REVIEW

**Comment by gilmoskowitz** [comment]
jenkins - retest this please, now that i've updated your task to create a .timezone file

**Comment by amikheychik** [comment]
@gilmoskowitz testing out what's missing in the current setup (@xtpclark said he used this branch)

1. Deployer user must have public and private keys. Public key is used as a deploy key on Github. (Important limitation: key must be unique across all the github repos, so at the moment it can be used only for xTC projects' repos).
2. In `${HOME}/.ssh` there is a directory `xtuple.xd` with `root` and `deployer` RSA key pairs: they shouldn't be there. `root` RSA should be for the `root` user, and `deployer` is `id_rsa` for `${DEPLOYER}` user.
3. `php-zip` extension is missing.
4. `/var/www/html` directory is still there.
5. `development`, `stage` and `production` DBs were missing.

I stopped after the last one, as didn't want to manually edit `pg_hda.conf` to finish the Drupal installation, so not sure if there are issues with Nginx setup.

Also, I have a major concern that we don't have a reasonable domain patterns structure, that would cover all our needs and would be consistent (and predictable) across the installations as it has always been with xTC installations before.

**Comment by amikheychik** [comment]
@gilmoskowitz one more issue:

```
PHP Warning:  date_default_timezone_get(): Invalid date.timezone value '{TIMEZONE}', we selected the timezone 'UTC' for now. in /var/www/stage/drupal/xdruple/src/Xtuple/Xdruple/Application/Console/Command/Install/InstallDrupalCommand.php on line 133
```

We might want to check all `php.ini` changes.

**Comment by gilmoskowitz** [comment]
nginx complains that
```
nginx: [emerg] "user" directive is not allowed here in /etc/nginx/sites-enabled/web:2
```
which comes from `templates/nginx/nginx-site` and nginx.sh:~153-159. What's the right behavior or template here?

**Comment by amikheychik** [comment]
@gilmoskowitz this is not an xTC-related template, maybe @xtpclark can help

**Comment by amikheychik** [comment]
@gilmoskowitz `environment.php` is not used and `environment.xml` that is used now should not be stored in xTAU, it should be coming from the project repo.

File: CreatePackages.sh [hunk 1]
```diff
@@ -250,24 +250,18 @@ xtc_build_xtuplecommerce_envphp() {
...
-  echo "Populating the environment.php file with settings for ${CRMACCT}"
-  echo "See loadcrm_gitconfig() and checkcrm_gitconfig()"
-  echo "Values are from ${HOME}/.gitconfig"
+  log "Populating environment.xml file with settings for ${CRMACCT}"
+  log "See loadcrm_gitconfig() and checkcrm_gitconfig()"
 
   echo "Writing out environment.xml"
   mkdir --parents ${SITE_WEBROOT}/application/config
-  safecp ${WORKDIR}/templates/application_environment.xml ${BUILD_XTC_CONF_DIR}/environment.xml
+  safecp ${SITE_WEBROOT}drupal/xdruple/dist/environment.xml.dist ${SITE_WEBROOT}/application/config/environment.xml
```
Comment (amikheychik) [inline CreatePackages.sh]: What is the difference between `xtc_build_xtuplecommerce_envphp()` and `setup_flywheel()`?

File: templates/xtau_config.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  "nginx": {
+    "DOMAIN_NAME":           [ "",     "" ],
+    "ERP_KEY_FILE_PATH":     [ "",     "Full path to encryption key" ],
+    "HOSTNAME":              [ "",     "" ],
+    "SERVER_CRT":            [ "",     "" ],
+    "SERVER_KEY":            [ "",     "" ]
+  },
+
+  "xtc": {
+    "COMMERCE_AUTHNET_AIM_LOGIN":           [ "",     "self-evident" ],
```
Comment (amikheychik) [inline templates/xtau_config.json]: If it's not used for ERP, it's not required here.

File: templates/xtau_config.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    "DOMAIN_ALIAS":          [ "",     "" ],
+    "ECOMM_DB_NAME":         [ "",     "Name of Drupal database" ],
+    "ECOMM_DB_USERNAME":     [ "",     "Name of Drupal database admin user" ],
+    "ECOMM_DB_USERPASS":     [ "",     "Password for Drupal database admin user" ],
+    "ECOMM_EMAIL":           [ "",     "email address for xTupleCommerce webmaster" ],
+    "ECOMM_SITE_NAME":       [ "",     "" ],
+    "ERP_APPLICATION":       [ "",     "Name of the Drupal application to be installed" ],
+    "ERP_DEBUG":             [ "false",  "true = disable SSL checking (**security hole**)" ],
+    "ERP_ISS":               [ "",     "" ],
+    "FEDEX_ACCOUNT_NUMBER":  [ "",     "self-evident" ],
```
Comment (amikheychik) [inline templates/xtau_config.json]: FedEx and UPS are configured in Drupal UI

File: templates/xtau_config.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    "LOGDIR":                [ "",     "**Parent** directory of website logs" ],
+    "MAX_EXECUTION_TIME":    [ "",     "PHP setting" ],
+    "SYSLOGID":              [ "",     "`SyslogIdentifier` used to set up xtuple-systemd.service" ],
+    "UPS_ACCESS_KEY":        [ "",     "self-evident" ],
+    "UPS_ACCOUNT_ID":        [ "",     "self-evident" ],
+    "UPS_PASSWORD":          [ "",     "self-evident" ],
+    "UPS_PICKUP_SCHEDULE":   [ "",     "self-evident" ],
+    "UPS_USER_ID":           [ "",     "self-evident" ],
+    "WEBROOT":               [ "",     "**Parent** directory of all websites" ],
+    "WORKFLOW_ENV":          [ "",     "Pantheon workflow, typically one of `dev`, `stage`, `live`" ]
```
Comment (amikheychik) [inline templates/xtau_config.json]: That's not a Pantheon workflow.

**Pull request merged**